datalad/interface/save.py

```diff
@@ -81,6 +83,26 @@ def __call__(message=None, files=None, dataset=None,
         if auto_add_changes:
             files = [ds.path]
 
+        # treat special case of still untracked subdatasets.
+        # those need to become submodules now, as they are otherwise added
+        # without an entry in .gitmodules, and subsequently break Git's
+        # submodule functionality completely
+        for utf in ds.repo.repo.untracked_files:
+            utf_abspath = opj(ds.path, utf)
+            if not isdir(utf_abspath):
+                # this cannot be a repository
+                continue
+
+            # test whether the potential submodule is scheduled for saving
+            if sum([realpath(utf_abspath).startswith(_with_sep(realpath(f)))
+                    for f in files]):
+                # matches at least one path -> turn into submodule
+                _install_subds_inplace(
+                    ds=ds,
+                    path=utf_abspath,  # can be ignored, we don't need the return value
+                    relativepath=utf,
+                    name=None)
```

Comment:
No, this would not work. If the untracked file is not covered by that list,
it would not be saved later on. Only list elements will be.

On Sep 8, 2016 2:50 PM, "Yaroslav Halchenko" <notifications@github.com>
wrote:

> In datalad/interface/save.py
> <https://github.com/datalad/datalad/pull/752#discussion_r77998567>:
>
> > +        # submodule functionality completely
> > +        for utf in ds.repo.repo.untracked_files:
> > +            utf_abspath = opj(ds.path, utf)
> > +            if not isdir(utf_abspath):
> > +                # this cannot be a repository
> > +                continue
> > +
> > +            # test whether the potential submodule is scheduled for saving
> > +            if sum([realpath(utf_abspath).startswith(_with_sep(realpath(f)))
> > +                    for f in files]):
> > +                # matches at least one path -> turn into submodule
> > +                _install_subds_inplace(
> > +                    ds=ds,
> > +                    path=utf_abspath,  # can be ignored, we don't need the return value
> > +                    relativepath=utf,
> > +                    name=None)
>
> so
>
> else:
>        raise RuntimeError("Michael didn't see how this could happen")
>
> ? ;)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/datalad/datalad/pull/752/files/f38eebf7993ba7dbcaa0cd4959d2e63c18cd82c3#r77998567>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAIVHwfG-iszjhpvFjm4VPAIOQsPhudHks5qoAR4gaJpZM4Jzb1B>
> .
>